internal/provider/resource_tfe_oauth_client.go

```diff
@@ -157,7 +158,7 @@ func resourceTFEOAuthClientCreate(d *schema.ResourceData, meta interface{}) erro
 	if serviceProvider == tfe.ServiceProviderAzureDevOpsServer {
 		options.PrivateKey = tfe.String(privateKey)
 	}
-	if serviceProvider == tfe.ServiceProviderBitbucketServer {
+	if serviceProvider == tfe.ServiceProviderBitbucketServer || serviceProvider == tfe.ServiceProviderBitbucketDataCenter {
```

Comment:
It looks like you cannot call this endpoint to create the service you're adding in this PR. See [docs](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/oauth-clients#create-an-oauth-client)

Reply:
I will double check which create endpoint is being called @laurenolivia . I was able to test the provider locally and create an `oauth_client` for a `bitbucket_data_center` service provider.
```terraform
resource "tfe_oauth_client" "test" {
  name = "bbdc-cli-tester"
  # organization     = "zain-tfe"
  api_url          = "https://bitbucket.dev.tf-vcs.hashicorp.services"
  http_url         = "https://bitbucket.dev.tf-vcs.hashicorp.services"
  key              = "my_key"
  secret           = "-----BEGIN RSA PRIVATE KEY-----XXXXX-----END RSA PRIVATE KEY-----\n"
  rsa_public_key   = "-----BEGIN PUBLIC KEY-----XXXXX-----END PUBLIC KEY-----\n"
  service_provider = "bitbucket_data_center"
}
```

Reply:
PR for terraform cloud api docs: https://github.com/hashicorp/terraform-docs-common/pull/624

Reply:
Cloud API docs approved. Thanks Lauren! 😄